src/services/Config.php

```diff
@@ -11,6 +12,11 @@ public function getConfigFromFile(string $filename): array
     {
         $overrides = [];
         $original = parent::getConfigFromFile($filename);
+        
+        // Convert fluent-style configs to an array
+        if ($original instanceof GeneralConfig) {
+            $original = collect($original)->toArray();
+        }
```

Comment:
This is nifty and definitely solves the problem. Thanks for figuring this out.

However, I think it'd be cleaner to add an early bail instead of forcing every config through this custom code. E.g., we'd need to move the original to the top, and add the early exit,

```php
$original = parent::getConfigFromFile($filename);
if (!is_array($original) || $filename !== 'app.web') {
  return $original;
}
```

then, below the bail we could assume we're operating on `app.web` (which doesn't have a fluent style config setup yet,

```
return array_merge($original, require __DIR__ . '/../config/app.web.php');
```

Make sense? I like this because it it's more clear that we're just not going to touch anything _other_ than `app.web`.

Reply:
I think I'm tracking with ya, @markhuot! Would the early exit look like this because `$original` isn't an array in a fluent-style config and needs to be coerced to one?

```php
public function getConfigFromFile(string $filename): array
{
    $original = parent::getConfigFromFile($filename);

    if (!is_array($original) || $filename !== 'app.web') {
        return collect($original)->toArray();
    }

    return array_merge($original, require __DIR__ . '/../config/app.web.php');
}
```

Reply:
In other words that `return collect($original)…` could probably just be `return $original;`.

Reply:
Ooooh—I'm following you, @markhuot. Okay, just modified this early return. Had to revise the function's return types to `array|BaseConfig` since it can be either/or now.

Reply:
Ah! This is great.

Just one issue on Craft 3.7, there’s no `BaseConfig` so we get a syntax error. Can you move the typehint to a docblock so it doesn’t break on 3.7 and I think that’ll do it!